---
title: "Cryptic expression"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyr)
library(ggpubr)
library(dplyr)
library(tidyverse)
library(repurrrsive)
library(jsonlite)
library(clipr)
library(naniar)
library(rstatix)
library(knitr)
library(snapcount)
library(ggsignif)
```

```{r function: count_query, include = FALSE}

count_query <- function(gene_name, snapcount_coords, strand_code) {
  query <-  QueryBuilder(compilation = 'tcga',regions = snapcount_coords) 
  
  if(strand_code == "+"){
    query <- set_row_filters(query, strand == "+") 
  }else if(strand_code == "-"){
    query <- set_row_filters(query, strand == "-") 
  }
  
  query_exact <- set_coordinate_modifier(query, Coordinates$Exact) 
  print("junction querying beginning")
  juncs_on <- query_jx(query) 
  juncs_on_flat <- query_jx(query,return_rse = FALSE) 
  samples_with <- juncs_on@colData |> 
    as.data.frame()
  print("junction querying done")
  samples_with <- samples_with |> 
    select(c("rail_id",
             "gdc_cases.demographic.gender",
             "gdc_cases.submitter_id",
             "gdc_cases.project.name",
             "gdc_cases.project.primary_site",
             "gdc_cases.diagnoses.tumor_stage",
             "gdc_cases.samples.sample_type",
             "cgc_case_primary_site",
             "junction_coverage",
             "junction_avg_coverage")) 
  
  juncs_on_flat <- juncs_on_flat |> 
    select(chromosome,start,end,strand,samples) |> 
    separate_rows(samples, sep = ',') |> 
    filter(samples != "") |> 
    separate(samples, into = c("rail_id","count")) |> 
    mutate(rail_id = as.integer(rail_id)) |> 
    mutate(count = as.numeric(count)) 
  
  gene_df <- juncs_on_flat |> 
    left_join(samples_with) |> 
    filter(end == str_split(snapcount_coords,'-',simplify = TRUE)[[2]]) |> 
    filter(start == str_split(str_split(snapcount_coords,'-',simplify = TRUE)[[1]],":",simplify = TRUE)[[2]])
  return(gene_df)
}

read_save_mutation_data <- function(filename) {
  df <- janitor::clean_names(read.csv(filename, sep = "\t"))
  return(df)
}
# read_save_mutation_data("TCGA-VQ-A91N_mutations.tsv") |> View()

combine_mutation_data <- function(folder_path, pattern = "^TCGA.*_mutations.tsv$") {
  mutation_files <- list.files(folder_path,
                               pattern = pattern,
                               full.names = TRUE)
  #print(mutation_files)
  df <- purrr::map(mutation_files, read_save_mutation_data)
  patient_id <- purrr::simplify(purrr::map(mutation_files, basename))
  patient_id = gsub("_mutations.tsv", "", patient_id)
  non_empty_df <- df[sapply(df, function(x) nrow(x) > 0)]
  non_empty_patient_id <- patient_id[sapply(df, function(x) nrow(x) > 0)]
  df <- purrr::map2(non_empty_df, non_empty_patient_id, ~cbind(.x, case_submitter_id = .y))
  df <- data.table::rbindlist(df)
  return(df)
}
```

```{r query (cryptic and anno), include = FALSE}
STMN2_cryptic_cBio <- read.table("STMN2_cryptic_cBio.txt", sep=",")
ARHGAP32_cryptic_cBio <- read.table("ARHGAP32_cryptic_cBio.txt", sep=",")
SYNJ2_cryptic_cBio <- read.table("SYNJ2_cryptic_cBio.txt", sep=",")
```

```{r any cases with all three or two cryptic events?, include = FALSE}
common_patients_STMN2_ARHGAP32 <- intersect(STMN2_cryptic_cBio$case_submitter_id, ARHGAP32_cryptic_cBio$case_submitter_id) 
      # 5 cases
intersect(STMN2_cryptic_cBio$case_submitter_id, SYNJ2_cryptic_cBio$case_submitter_id) # none
common_patients_ARHGAP32_SYNJ2 <- intersect(ARHGAP32_cryptic_cBio$case_submitter_id, SYNJ2_cryptic_cBio$case_submitter_id) 
      # 31 cases with both ARHGAP32 and SYNJ2 cryptic events
```

```{r df with ARHGAP32 SYNJ2 common cases, echo = FALSE}
common_ARHGAP32_SYNJ2 <- ARHGAP32_cryptic_cBio |> 
  filter(case_submitter_id %in% common_patients_ARHGAP32_SYNJ2) |> 
  select(case_submitter_id, start, end, strand, anno_count, cryptic_count, jir) |> 
  left_join(SYNJ2_cryptic_cBio, by = "case_submitter_id", suffix = c("_ARHGAP32","_SYNJ2")) |> 
  distinct() |> 
  drop_na() |> 
  select(-cancer) |> 
  filter(gdc_cases_samples_sample_type == "Primary Tumor") |> 
  janitor::clean_names() 

  # within each patient, what is the fraction of cryptic to annotated reads?
common_ARHGAP32_SYNJ2 |> 
  select(case_submitter_id,gdc_cases_project_primary_site,disease_survival_months,mutation_count,jir_arhgap32,jir_synj2) |> 
  pivot_longer(cols = c(jir_arhgap32,jir_synj2),
               names_to = "gene",
               values_to = "cryptic_jir") |>  
  mutate(gene = str_replace(gene, "jir_", "")) |> 
  mutate(cryptic_jir = 1 - cryptic_jir) |> 
  ggplot(aes(x = gene,y = cryptic_jir)) + 
  geom_boxplot()

  # within each patient, is ARHGAP32 or SYNJ2 cryptic expressed at a higher rate?
common_ARHGAP32_SYNJ2 |> 
  select(case_submitter_id,gdc_primary_site,disease_survival_months,mutation_count,jir_arhgap32,jir_synj2) |> 
  pivot_longer(cols = c(jir_arhgap32,jir_synj2),
               names_to = "gene",
               values_to = "cryptic_jir") |>  
  mutate(gene = str_replace(gene, "jir_", "")) |> 
  mutate(cryptic_jir = 1 - cryptic_jir) |> 
  ggplot(aes(x = gene,y = cryptic_jir,group = case_submitter_id)) + 
  geom_point() +
  geom_line()

common_ARHGAP32_SYNJ2 |> 
  select(case_submitter_id,gdc_primary_site,disease_survival_months,mutation_count,jir_arhgap32,jir_synj2) |> 
  mutate(jir_arhgap32 = 1 - jir_arhgap32 , jir_synj2 = 1 - jir_synj2) |> 
  mutate(arh_ratio = log2(jir_arhgap32 / jir_synj2)) |> 
  ggplot(aes(y = arh_ratio,x = 1)) + 
  geom_violin() + 
  geom_hline(yintercept = 0) + 
  ylab("Log2Fold Ratio of ARHGAP32 to SYNJ2 cryptic expression")
```


```{r df with STMN2 ARHGAP32 common cases, include = FALSE}
common_STMN2_ARHGAP32 <- STMN2_cryptic_cBio |> 
  filter(gdc_cases_samples_sample_type == "Primary Tumor",
         case_submitter_id %in% common_patients_STMN2_ARHGAP32) |> 
  select(case_submitter_id, cryptic_count, anno_count, jir) |> 
  left_join(ARHGAP32_cryptic_cBio, by = "case_submitter_id",suffix = c("_STMN2","_ARHGAP32")) |>   
  distinct() |> 
  filter(gdc_cases_samples_sample_type == "Primary Tumor") |> 
  janitor::clean_names() |> 
  select(-c(chromosome, start, end, strand, rail_id, junction_coverage, junction_avg_coverage)) 
```


```{r df with all common cases, echo = FALSE}
all_common_cases <- common_STMN2_ARHGAP32 |> 
  full_join(common_ARHGAP32_SYNJ2) 

all_common_cases |> 
  pivot_longer(cols = c(jir_stmn2, jir_arhgap32, jir_synj2),
               names_to = "gene",
               values_to = "cryptic_jir") |> 
  mutate(gene = str_replace(gene, "jir_", "")) |> 
  mutate(cryptic_jir = 1 - cryptic_jir) |> 
  ggplot(aes(x = gene, 
             y = cryptic_jir,
             group = case_submitter_id)) +
  geom_point() +
  geom_line()
```

```{r case set of all 14 common patients, include = FALSE}

all_common_cases <- all_common_cases |> 
  mutate(case_id = paste0(study_id, ":", case_submitter_id)) |> 
  distinct()

#write_clip(unique(all_common_cases$case_id)) # copies all common patients to clipboard
#write_clip(unique(all_common_cases$case_submitter_id))

# filter mutation_clinical_data for 14 common patients only 

#x <- mutation_clinical_data |> 
  #filter(case_submitter_id %in% all_common_cases)

#intersect(all_common_cases$case_submitter_id, mutation_clinical_data$case_submitter_id)


#intersect(STMN2_cryptic_cBio$case_submitter_id, SYNJ2_cryptic_cBio$case_submitter_id)
```