---
title: "STMN2"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(repurrrsive)
library(jsonlite)
library(clipr)
library(naniar)
library(ggpubr)
library(rstatix)
library(ggplot2)
library(tidyr)
library(dplyr)
library(knitr)
library(snapcount)
library(ggsignif)
```

Gene name = "STMN2" 
STMN2 cryptic coordinates = "chr8:79611215-79616821"
STMN2 annotated coordinates = "chr8:79611215-79636801"
STMN2 strand code = "+"

```{r query (cryptic and anno), include = FALSE}
STMN2_cryptic_query <- count_query(gene_name = "STMN2", snapcount_coords = "chr8:79611215-79616821", strand_code = "+")
STMN2_anno_query <- count_query(gene_name = "STMN2", snapcount_coords = "chr8:79611215-79636801", strand_code = "+")
```

```{r join cryptic and anno df, add jir column, include = FALSE}
STMN2_cryptic_query <- STMN2_cryptic_query |> rename("cryptic_count" = "count")
STMN2_anno_query <- STMN2_anno_query |> rename("anno_count" = "count")

STMN2_query <- STMN2_cryptic_query |> 
  select(gdc_cases.submitter_id, cryptic_count) |> 
  left_join(STMN2_anno_query, by = "gdc_cases.submitter_id")

STMN2_query <- STMN2_query |> 
  relocate(cryptic_count, .after = anno_count)

STMN2_query <- STMN2_query |> 
  mutate(jir = anno_count/(anno_count + cryptic_count)) |> 
  relocate(jir, .after = cryptic_count)

#write_clip(STMN2_query$gdc_cases.submitter_id)
```

# Cancers with STMN2 expression

```{r Import clinical data from TCGA, include = FALSE}
STMN2_clinical_orig <- read.csv("STMN2_clinical.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)
head(STMN2_clinical_orig)

STMN2_clinical <- STMN2_clinical_orig

STMN2_clinical <- STMN2_clinical|> 
  select(case_submitter_id, project_id, age_at_index, ethnicity, gender, race, ajcc_pathologic_stage) |> 
  separate(project_id, into = c("project", "cancer_type")) |> 
  select(-project)
```

```{r bar plot - cancer types with STMN2 events}
STMN2_clinical |> 
  ggplot(aes(x = fct_rev(fct_infreq(cancer_type)))) +
  geom_bar(aes(fill = cancer_type)) +
  coord_flip() +
  labs(
    x = "Cancer Type",
    y = "Number of Cases",
    title = "STMN2 is expressed in mostly breast and brain cancers" 
  ) +
  theme(legend.position = "none", plot.title = element_text(size=10)) 
```
**Figure 1**: STMN2 events are found mostly in breast and brain cancer patients. 
BRCA = breast cancer; LGG = low-grade gliomas (brain tumours)

# Cancers with cryptic STMN2 expression

```{r joining jir table with clinical table, include = FALSE}
jir_new_STMN2 <- jir_new_STMN2 |> rename("case_submitter_id" = "gdc_cases.submitter_id")

STMN2_clinical_jir <- jir_new_STMN2 |> 
  left_join(STMN2_clinical, by=c("case_submitter_id"))

STMN2_clinical_jir <- STMN2_clinical_jir |> 
  select(-gdc_cases.diagnoses.tumor_stage, -age_at_index, -gdc_cases.demographic.gender, 
         -ajcc_pathologic_stage, -ethnicity, -race, -cancer_type, -gender)
 
STMN2_clinical_jir <- STMN2_clinical_jir |> 
  rename("cancer_type" = "gdc_cases.project.name") |> 
  rename("gdc_primary_site" = "gdc_cases.project.primary_site") |> 
  rename("sample_type" = "gdc_cases.samples.sample_type") |> 
  rename("cgc_primary_site" = "cgc_case_primary_site")
```

```{r filter for cryptic coverage 2}
STMN2_clinical_jir_cryptic <- STMN2_clinical_jir |> 
  filter(STMN2_cryptic_coverage > 2) 
```

## Primary sites of cancers 

```{r bar plot - primary sites of cancers with cryptic STMN2 events}
STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  ggplot(aes(x = fct_rev(fct_infreq(gdc_primary_site)))) +
  geom_bar(aes(fill = gdc_primary_site)) +
  coord_flip() +
  labs(
    x = "Primary Site of Cancer",
    y = "Number of Cases",
    title = 
  "Primary sites of cancers with cryptic STMN2 events are mostly the adrenal gland and brain" 
  ) +
  theme(legend.position = "none", plot.title = element_text(size=9)) 
```
**Figure 2**: Cryptic STMN2 events are found mostly in the adrenal gland and brain.

Interestingly, Figure 2 shows that cryptic STMN2 expression is low in the breast. Comparing this to Figure 1, this shows that the STMN2 events expressed in breast cancers may be mostly annotated non-cryptic events. 

## Cryptic STMN2 junction coverage in different sites of cancers

```{r boxplot - overall STMN2 junction coverage in different cancer sites, warning = FALSE}
STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  filter(cgc_primary_site != "") |> 
  ggplot(aes(x = junction_avg_coverage, y = fct_reorder(gdc_primary_site, 
                                                  junction_avg_coverage, median))) +
  geom_boxplot(aes(fill = gdc_primary_site)) +
  labs(
    x = "Junction Average Coverage",
    y = "Primary Site of Cancer",
  ) +
  theme(legend.position = "none", plot.title = element_text(size=10)) +
  geom_signif(comparisons = list(c("Stomach", "Breast"), c("Stomach", "Brain")),
              map_signif_level = TRUE,
              y_position = c(75, 80))
```
**Figure 3**: Stomach and breast cancers are the most deeply sequenced.

```{r boxplot - cryptic STMN2 expression in different cancer sites}
STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  ggplot(aes(x = STMN2_cryptic_coverage, 
             y = fct_reorder(gdc_primary_site, STMN2_cryptic_coverage, median))) +
  geom_boxplot(aes(fill = gdc_primary_site)) +
  labs(
    x = "STMN2 Cryptic Coverage",
    y = "Primary Site of Cancer",
  ) +
  theme(legend.position = "none", plot.title = element_text(size=10))
```
**Figure 4**: On average, cancers in the stomach and breast have the greatest number of reads supporting cryptic STMN2 events. Cancers of the brain have significantly fewer reads surpporting cryptic events. 


## Which cancers have the most cryptic STMN2 events?

```{r number / fraction of cases with STMN2 events in different cancers, echo = FALSE}
STMN2_events_different_cancers <- STMN2_clinical_jir_cryptic |> 
  drop_na() |>
  janitor::tabyl(cancer_type) |> arrange(-percent)

kable(STMN2_events_different_cancers, 
      caption = "Breast and brain cancers have high cryptic STMN2 expression.")
```

## Where are the cancers with cryptic STMN2 events located?

```{r number / fraction of cases with STMN2 events in different primary cancer sites, echo = FALSE}
STMN2_events_primary_sites1 <- STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  janitor::tabyl(gdc_primary_site) |> arrange(-percent)

kable(STMN2_events_primary_sites1, caption = "Cancers with cryptic STMN2 events are found primarily in the adrenal gland and brain")
```

# cBioPortal Clinical Data

```{r data import - cBioPortal Clinical Data, include = FALSE}

cBio_all_clinical_orig <- read.csv("cBio_clinical_data.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)

cBio_clinical <- cBio_all_clinical_orig

cBio_clinical <- cBio_clinical |> 
  select(Study.ID, Patient.ID, Sample.ID, Aneuploidy.Score, Cancer.Type, TCGA.PanCanAtlas.Cancer.Type.Acronym, Cancer.Type.Detailed,
         Months.of.disease.specific.survival, Mutation.Count, Overall.Survival..Months.)

cBio_clinical <- cBio_clinical |> 
  rename("case_submitter_id" = "Patient.ID") |> 
  rename("cancer" = "Cancer.Type.Detailed") |> 
  rename("cancer_abbrev" = "TCGA.PanCanAtlas.Cancer.Type.Acronym")
```

```{r join cBio clinical data with STMN2 cryptic table, include = FALSE}
STMN2_cryptic_cBio <- STMN2_clinical_jir_cryptic |> 
  left_join(cBio_clinical, by = "case_submitter_id") 

STMN2_cryptic_cBio <- STMN2_cryptic_cBio |> 
   select(-cgc_primary_site, -Cancer.Type, -Sample.ID, -Overall.Survival..Months.) |>
  rename("disease_survival_months" = "Months.of.disease.specific.survival") |> 
  relocate(case_submitter_id, .after = sample_id) |> 
  relocate(cancer_type, .after = case_submitter_id) |> 
  relocate(gdc_primary_site, .after = cancer_abbrev) |> 
  relocate(sample_type, .after = gdc_primary_site) |> 
  janitor::clean_names()

```

## Mutation Counts

```{r calculate mean n mutations for each cancer type, include = FALSE}
STMN2_cryptic_cBio_mutations <- STMN2_cryptic_cBio |> 
  drop_na() |> 
  filter(mutation_count !="NA") |> 
  group_by(cancer_type) |> 
  mutate(mean_mutation_count = mean(as.numeric(mutation_count))) |> 
  ungroup() |> 
  select(cancer_type, cancer_abbrev, mean_mutation_count) |> 
  unique() 
```

```{r bar chart - n mutations for each cancer type}
STMN2_cryptic_cBio_mutations |> 
  ggplot(aes(x = fct_reorder(cancer_abbrev, mean_mutation_count, median), y = mean_mutation_count)) +
  geom_bar(stat = 'identity', aes(fill = cancer_abbrev)) +
  labs(
    x = "Cancer Type",
    y = "Mean Mutation Count",
    title = "Glioblastoma Multiforme has the greatest number of mutations"
  ) +
  theme(legend.position = "none")
```

There is an outlier in GBM for mutation count, which heavily skews the mean for this cancer type. 

```{r boxplot - n mutations for each cancer type}
STMN2_cryptic_cBio <- STMN2_cryptic_cBio |> 
  mutate_at("mutation_count", as.numeric) 

STMN2_cryptic_cBio |> 
  drop_na() |> 
  filter(mutation_count < 2500) |> 
  ggplot(aes(x = fct_reorder(cancer_abbrev, mutation_count, median), 
             y = mutation_count)) +
  geom_boxplot() +
  labs(
    x = "Cancer Type",
    y = "Mutation Count",
    title = "UCEC and LUSC cancers have the greatest number of mutations"
  ) +
  geom_signif(comparisons = list(c("LUSC", "UCEC")),
map_signif_level = TRUE,
y_position = c(1000))
```

**Figure 5**: Among cancer patients with STMN2 cryptic expression, uterine corpus endometrial carcinoma (UCEC) has the greatest number of mutations.

## Mutation Data of one patient

```{r data import, include = FALSE}
patient_mutations_orig <- read.csv("TCGA-A5-A0G2_mutations.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE) 

patient_mutations <- patient_mutations_orig |> 
  select(Gene, Protein.Change, Annotation, Functional.Impact, Chromosome, Start.Pos, End.Pos, Ref, Var, HGVSg, Mutation.Type, Variant.Type, Allele.Freq)
```

```{r number of mutations in each gene}
patient_mutations |> 
  count(Gene, sort = TRUE) |> 
  filter(n > 17) |> 
  ggplot(aes(x = fct_reorder(Gene, n, mean), y = n)) +
  geom_bar(stat = 'identity', aes(fill = Gene)) +
  labs(
    x = "Gene",
    y = "Mutation Count"
  ) +
  theme(legend.position = "none")
```

**Figure 6**: Among cancer patients with STMN2 cryptic expression, the titin gene has the greatest number of mutations.

```{r types of mutations in TTN}
patient_mutations |> 
  filter(Gene == "TTN") |> 
  n_distinct()

patient_mutations |> 
  filter(Gene == "TTN") |> 
  unique() |> 
  ggplot(aes(x = fct_rev(fct_infreq(Mutation.Type)))) +
  geom_bar(aes(fill = Mutation.Type)) +
  labs(
    x = "Mutation Type",
    y = "Count in TTN Gene"
  ) +
  theme(legend.position = "none")
```

**Figure 7**: Most of the mutations seen in the titin gene are missense mutations.

## STMN2 cryptic expression 

```{r STMN2 cryptic coverage vs mutation count, warning = FALSE}

STMN2_cryptic_cBio <- STMN2_cryptic_cBio |> 
  mutate_at("mutation_count", as.numeric) 

STMN2_cryptic_cBio |> 
  drop_na() |> 
  filter(mutation_count < 400, stmn2_cryptic_coverage > 1) |> 
  ggplot(aes(x = as.factor(stmn2_cryptic_coverage), y = mutation_count)) +
  geom_boxplot() +
  labs(
    x = "Number of STMN2 cryptic events",
    y = "Mutation Count"
  ) +
  geom_signif(comparisons = list(c("6", "8"), c("5", "6"), c("3", "6")),
              map_signif_level = TRUE,
              y_position = c(250, 300, 200))
```

```{r}
STMN2_cryptic_cBio |> 
  drop_na() |> 
  filter(mutation_count < 2500) |> 
  ggplot(aes(x = stmn2_cryptic_coverage, y = mutation_count)) +
  geom_hex() +
  labs(
    x = "Number of STMN2 cryptic events",
    y = "Mutation Count"
  )
```

**Figure 8**: There is no correlation between number of STMN2 cryptic events and number of mutations.

# Cancer Driver Genes

```{r COSMIC Cancer Gene Consensus, include = FALSE}
cosmic_cancer_genes <- read.csv("cosmic_cancer_gene_consensus.csv") |> 
  rename("Gene" = "Gene.Symbol")

# filter for cancer driver genes

cosmic_patient_mutations <- filter(patient_mutations, Gene %in% cosmic_cancer_genes$Gene)
```

## Mutations in each cancer gene

```{r number of mutations in each cancer gene}
cosmic_patient_mutations |> 
  count(Gene, sort = TRUE) |> 
  filter(n > 11) |> 
  ggplot(aes(x = fct_reorder(Gene, n, mean), y = n)) +
  geom_bar(stat = 'identity', aes(fill = Gene)) +
  labs(
    x = "Cancer Gene",
    y = "Mutation Count"
  ) +
  theme(legend.position = "none")
```

**Figure 9**: Ovarian cancer gene MUC16 has the greatest number of mutations.

```{r types of mutations in MUC16}
cosmic_patient_mutations |> 
  filter(Gene == "MUC16") |> 
  ggplot(aes(x = fct_rev(fct_infreq(Mutation.Type)))) +
  geom_bar(aes(fill = Mutation.Type)) +
  labs(
    x = "Mutation Type",
    y = "Count in MUC16 Gene"
  ) +
  theme(legend.position = "none")
```

**Figure 10**: Ovarian cancer gene MUC16 has mostly missense mutations.

## Types of cancer genes

```{r types of cancer genes, include = FALSE}
cosmic_patient_mutations <- cosmic_patient_mutations |> 
  left_join(cosmic_cancer_genes, by = "Gene") 

cosmic_patient_mutations <- cosmic_patient_mutations |> 
  select(-Entrez.GeneId, -Tier, -Hallmark, -Chr.Band, -Somatic, -Germline, -Tumour.Types.Somatic., 
         -Tumour.Types.Germline., -Cancer.Syndrome, -Molecular.Genetics, -Translocation.Partner, 
         -Other.Germline.Mut, -Other.Syndrome, -Synonyms) 

  # fusion
      # Gene fusions, or translocations, resulting from chromosomal rearrangements are the most common mutation class. 
      #They lead to chimeric transcripts or to deregulation of genes through juxtapositioning of novel promoter or enhancer regions.

  # TSG = tumour suppressor gene

  # oncogene

cosmic_patient_mutations <- cosmic_patient_mutations |> 
  mutate(role_in_cancer = case_when(
           (grepl("TSG", Role.in.Cancer)) ~  "TSG",
            (grepl("oncogene", Role.in.Cancer)) ~ "oncogene",
             Role.in.Cancer == "fusion" ~ 'fusion'))
```

```{r}
cosmic_patient_mutations |> 
  drop_na() |> 
  ggplot(aes(x = role_in_cancer)) +
  geom_bar() +
  labs(
    x = "Role of the gene in cancer",
    y = "Number of mutations",
    title = "Most of the mutations are found in tumour suppressor genes"
  ) +
  theme(legend.position = "none", plot.title = element_text(size=12))
```

**Figure 11**: Among the mutations found in cancer patients with cryptic STMN2 expression, majority of the mutations are in tumour suppressor genes.

```{r types, echo = FALSE}
n_role_in_cancer <- cosmic_patient_mutations |> 
  drop_na() |> 
  janitor::tabyl(role_in_cancer) |> arrange(-percent)

kable(n_role_in_cancer, caption = "Almost half of the cancer genes mutated in patients with enrichment of cryptic STMN2 events are tumour suppressor genes.")
```

```{r COSMIC all mutations in census genes, include = FALSE}

cosmic_all_mutations_census_genes_orig <- read.csv("cosmic_all_mutations_in_census_genes.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)
cosmic_all_mutations_census_genes <- cosmic_all_mutations_census_genes_orig
cosmic_all_mutations_census_genes <- cosmic_all_mutations_census_genes |> filter(grepl("TCGA", Sample.name)) |> 
  select(-Site.subtype.1, -Site.subtype.2, -Site.subtype.3, -Histology.subtype.1, -Histology.subtype.2, -Histology.subtype.3, 
         -Genome.wide.screen, -GENOMIC_MUTATION_ID, -LEGACY_MUTATION_ID, -MUTATION_ID, -Mutation.zygosity, -LOH, -GRCh,
         -Mutation.somatic.status, -Pubmed_PMID, -ID_STUDY, -Sample.Type, -Tumour.origin, -Tier, -HGVSP, -HGVSC, -HGVSG, -Accession.Number)
cosmic_all_mutations_census_genes <- cosmic_all_mutations_census_genes |> rename("case_submitter_id" = "Sample.name")

#removing end digits from case_submitter_id
cosmic_all_mutations_census_genes$case_submitter_id <- ifelse(grepl("^(\\w+-){2}\\w+$", 
                                                                    cosmic_all_mutations_census_genes$case_submitter_id),
                                                              cosmic_all_mutations_census_genes$case_submitter_id,
                                                              sub("-\\d+$", "",
                                                                  cosmic_all_mutations_census_genes$case_submitter_id))

# left-join COSMIC all mutations to jir_cryptic ---------------------------

STMN2_cryptic_patients_mutations <- cosmic_all_mutations_census_genes |> 
  left_join(STMN2_clinical_jir_cryptic, by = "case_submitter_id")
```

```{r patients w cryptic vs patients w/o cryptic - mutations and genes}

```
