---
title: "Clinical, gene and mutation data of cancers with cryptic STMN2 events"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(repurrrsive)
library(jsonlite)
library(clipr)
library(naniar)
library(ggpubr)
library(rstatix)
library(ggplot2)
library(tidyr)
library(dplyr)
library(knitr)
library(snapcount)
library(ggsignif)
```

```{r snapcount_STMN2.R, include = FALSE}
gene_name = "STMN2" 
snapcount_coords_cryptic_STMN2 = "chr8:79611215-79616821"
snapcount_coords_annotated_STMN2 = "chr8:79611215-79636801"
strand_code = "+"

cryptic_query <-  QueryBuilder(compilation = 'tcga',regions = snapcount_coords_cryptic_STMN2) 

if(strand_code == "+"){
  cryptic_query <- set_row_filters(cryptic_query, strand == "+") 
}else if(strand_code == "-"){
  cryptic_query <- set_row_filters(cryptic_query, strand == "-") 
}

juncs_on_cryptic <- query_jx(cryptic_query) 
juncs_on_cryptic_flat <- query_jx(cryptic_query,return_rse = FALSE) 
samples_with_cryptic <- juncs_on_cryptic@colData |> 
  as.data.frame() 

samples_with_cryptic <- samples_with_cryptic |> 
  select(c("rail_id",
           "gdc_cases.demographic.gender",
           "gdc_cases.submitter_id",
           "gdc_cases.project.name",
           "gdc_cases.project.primary_site",
           "gdc_cases.diagnoses.tumor_stage",
           "gdc_cases.samples.sample_type",
           "cgc_case_primary_site",
           "junction_coverage", 
           "junction_avg_coverage")) 

juncs_on_cryptic_flat <- juncs_on_cryptic_flat |> 
  select(chromosome,start,end,strand,samples) |> 
  separate_rows(samples, sep = ',') |> 
  filter(samples != "") |> 
  separate(samples, into = c("rail_id","count")) |> 
  mutate(rail_id = as.integer(rail_id)) |> 
  mutate(count = as.numeric(count)) 

juncs_on_cryptic_flat |> 
  left_join(samples_with_cryptic) |> 
  filter(end == 79616821)

anno_query <-  QueryBuilder(compilation = 'tcga',regions = snapcount_coords_annotated_STMN2)
anno_query <- set_coordinate_modifier(anno_query, Coordinates$Exact) 
if(strand_code == "+"){
  anno_query <- set_row_filters(anno_query, strand == "+") 
}else if(strand_code == "-"){
  anno_query <- set_row_filters(anno_query, strand == "-")
}

cryptic_query_exact <- set_coordinate_modifier(cryptic_query, Coordinates$Exact) 

jir <- junction_inclusion_ratio(list(cryptic_query_exact),
                                list(anno_query),
                                group_names=c(glue::glue("{gene_name}_cryptic"),
                                              glue::glue("{gene_name}_annotated")))

jir_new <- jir |> 
  left_join(samples_with_cryptic,by = c('sample_id' = 'rail_id')) 
```


```{r GDC case IDs, include = FALSE}
jir_new$gdc_cases.submitter_id
#write_clip(jir_new$gdc_cases.submitter_id)
```

```{r Import gene data from TCGA, include = FALSE}
STMN2_events_genes_orig <- read.csv("STMN2_events_genes.tsv", sep = "\t", header = TRUE, na.strings="", fill = TRUE)
head(STMN2_events_genes_orig)

STMN2_events_genes <- STMN2_events_genes_orig

STMN2_events_genes <- STMN2_events_genes |> 
  rename("affected_cases_in_cohort" = "X..SSM.Affected.Cases.in.Cohort") |> 
    # number of cases where gene is mutated / number of cases tested for simple somatic mutations (3,259)
  rename("affected_cases_in_GDC" = "X..SSM.Affected.Cases.Across.the.GDC") |> 
    # number of cases where gene contains simple somatic mutations / number of cases tested for simple somatic mutations portal wide (13,714)
  rename("CNV_gain" = "X..CNV.Gain") |> 
    # number of cases where CNV gain observed in gene / number of cases tested for copy number alteration in gene (3,344)
  rename("CNV_loss" = "X..CNV.Loss") |> 
    # number of cases where CNV loss observed in gene / number of cases tested for copy number alteration in gene (3,344)
  rename("mutations" = "X..Mutations")
    # unique simple somatic mutations in the gene in cohort 
```


```{r Import mutation data from TCGA, include = FALSE}
STMN2_events_mutations_orig <- read.csv("STMN2_events_mutations.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)
head(STMN2_events_mutations_orig)

STMN2_events_mutations <- STMN2_events_mutations_orig
```

# Cancers with STMN2 expression

```{r Import clinical data from TCGA, include = FALSE}
STMN2_clinical_orig <- read.csv("STMN2_clinical.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)
head(STMN2_clinical_orig)

STMN2_clinical <- STMN2_clinical_orig

STMN2_clinical <- STMN2_clinical|> 
  select(case_submitter_id, project_id, age_at_index, ethnicity, gender, race, ajcc_pathologic_stage) |> 
  separate(project_id, into = c("project", "cancer_type")) |> 
  select(-project)
```


```{r bar plot - cancer types with cryptic STMN2 events}
STMN2_clinical |> 
  ggplot(aes(x = fct_rev(fct_infreq(cancer_type)))) +
  geom_bar(aes(fill = cancer_type)) +
  coord_flip() +
  labs(
    x = "Cancer Type",
    y = "Number of Cases",
    title = "STMN2 is expressed in mostly breast and brain cancers" 
  ) +
  theme(legend.position = "none", plot.title = element_text(size=10)) 
```
**Figure 1**: STMN2 events are found mostly in breast and brain cancer patients. 
BRCA = breast cancer; LGG = low-grade gliomas (brain tumours)

# Cancers with cryptic STMN2 expression

```{r joining jir table with clinical table, include = FALSE}
jir_new <- jir_new |> 
  rename("case_submitter_id" = "gdc_cases.submitter_id")

STMN2_clinical_jir <- jir_new |> 
  left_join(STMN2_clinical, by=c("case_submitter_id"))

STMN2_clinical_jir <- STMN2_clinical_jir |> 
  select(-gdc_cases.diagnoses.tumor_stage, -age_at_index, -gdc_cases.demographic.gender, 
         -ajcc_pathologic_stage, -ethnicity, -race, -cancer_type, -gender)
 
STMN2_clinical_jir <- STMN2_clinical_jir |> 
  rename("cancer_type" = "gdc_cases.project.name") |> 
  rename("gdc_primary_site" = "gdc_cases.project.primary_site") |> 
  rename("sample_type" = "gdc_cases.samples.sample_type") |> 
  rename("cgc_primary_site" = "cgc_case_primary_site")
```

```{r filter for cryptic coverage 2}
STMN2_clinical_jir_cryptic <- STMN2_clinical_jir |> 
  filter(STMN2_cryptic_coverage > 2) 
```

## Primary sites of cancers 

```{r bar plot - primary sites of cancers with cryptic STMN2 events}
STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  ggplot(aes(x = fct_rev(fct_infreq(gdc_primary_site)))) +
  geom_bar(aes(fill = gdc_primary_site)) +
  coord_flip() +
  labs(
    x = "Primary Site of Cancer",
    y = "Number of Cases",
    title = 
  "Primary sites of cancers with cryptic STMN2 events are mostly the adrenal gland and brain" 
  ) +
  theme(legend.position = "none", plot.title = element_text(size=9)) 
```
**Figure 2**: Cryptic STMN2 events are found mostly in the adrenal gland and brain.

Interestingly, Figure 2 shows that cryptic STMN2 expression is low in the breast. Comparing this to Figure 1, this shows that the STMN2 events expressed in breast cancers may be mostly annotated non-cryptic events. 

## Cryptic STMN2 junction coverage in different sites of cancers

```{r boxplot - overall STMN2 junction coverage in different cancer sites, warning = FALSE}
STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  filter(cgc_primary_site != "") |> 
  ggplot(aes(x = junction_avg_coverage, y = fct_reorder(gdc_primary_site, 
                                                        junction_avg_coverage, median))) +
  geom_boxplot(aes(fill = gdc_primary_site)) +
  labs(
    x = "Junction Average Coverage",
    y = "Primary Site of Cancer",
  ) +
  theme(legend.position = "none", plot.title = element_text(size=10)) +
  geom_signif(comparisons = list(c("Stomach", "Breast"), c("Stomach", "Brain")),
              map_signif_level = TRUE,
              y_position = c(75, 80))
```
**Figure 3**: Stomach and breast cancers are the most deeply sequenced.

```{r boxplot - cryptic STMN2 expression in different cancer sites}
STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  filter(cgc_primary_site != "") |> 
  ggplot(aes(x = STMN2_cryptic_coverage, 
             y = fct_reorder(gdc_primary_site, STMN2_cryptic_coverage, median))) +
  geom_boxplot(aes(fill = gdc_primary_site)) +
  labs(
    x = "Junction Average Coverage",
    y = "Primary Site of Cancer",
  ) +
  theme(legend.position = "none", plot.title = element_text(size=10))
```

On average, cancers in the stomach and breast have the greatest number of reads supporting cryptic STMN2 events. Cancers of the brain have significantly fewer reads surpporting cryptic events. 



## Which cancers have the most cryptic STMN2 events?

```{r number / fraction of cases with STMN2 events in different cancers, echo = FALSE}
STMN2_events_different_cancers <- STMN2_clinical_jir_cryptic |> 
  drop_na() |>
  janitor::tabyl(cancer_type) |> arrange(-percent)

kable(STMN2_events_different_cancers, 
      caption = "Breast and brain cancers have high cryptic STMN2 expression.")
```

## Where are the cancers with cryptic STMN2 events located?

```{r number / fraction of cases with STMN2 events in different primary cancer sites, echo = FALSE}
STMN2_events_primary_sites1 <- STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  janitor::tabyl(gdc_primary_site) |> arrange(-percent)

kable(STMN2_events_primary_sites1, caption = "Cancers with cryptic STMN2 events are found primarily in the adrenal gland and brain")
```

# TCGA Clinical Data

```{r data import - TCGA Clinical Data, include = FALSE}
TCGA_all_clinical_orig <- read.csv("TCGA_all_clinical.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)
head(TCGA_all_clinical_orig)

TCGA_clinical <- TCGA_all_clinical_orig

TCGA_clinical <- TCGA_clinical|> 
  select(case_submitter_id, project_id, age_at_index, ethnicity, gender, race, ajcc_pathologic_stage) |> 
  separate(project_id, into = c("project", "cancer_type")) |> 
  select(-project)
```

```{r data import - cBioPortal Clinical Data, include = FALSE}

cBio_all_clinical_orig <- read.csv("cBio_clinical_data.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)

cBio_clinical <- cBio_all_clinical_orig

cBio_clinical <- cBio_clinical |> 
  select(Study.ID, Patient.ID, Sample.ID, Aneuploidy.Score, Cancer.Type, TCGA.PanCanAtlas.Cancer.Type.Acronym, Cancer.Type.Detailed,
         Months.of.disease.specific.survival, Mutation.Count, Overall.Survival..Months.)

cBio_clinical <- cBio_clinical |> 
  rename("case_submitter_id" = "Patient.ID") |> 
  rename("cancer" = "Cancer.Type.Detailed") |> 
  rename("cancer_abbrev" = "TCGA.PanCanAtlas.Cancer.Type.Acronym")
```

```{r join cBio clinical data with STMN2 cryptic table, include = FALSE}
STMN2_cryptic_cBio <- STMN2_clinical_jir_cryptic |> 
  left_join(cBio_clinical, by = "case_submitter_id") 

STMN2_cryptic_cBio <- STMN2_cryptic_cBio |> 
   select(-cgc_primary_site, -Cancer.Type, -Sample.ID, -Overall.Survival..Months.) |> 
  rename("disease_survival_months" = "Months.of.disease.specific.survival") |> 
  relocate(case_submitter_id, .after = sample_id) |> 
  relocate(cancer_type, .after = case_submitter_id) |> 
  relocate(gdc_primary_site, .after = cancer_abbrev) |> 
  relocate(sample_type, .after = gdc_primary_site) |> 
  janitor::clean_names()

```

## Mutation Counts

```{r calculate mean n mutations for each cancer type, include = FALSE}
STMN2_cryptic_cBio_mutations <- STMN2_cryptic_cBio |> 
  drop_na() |> 
  filter(mutation_count !="NA") |> 
  group_by(cancer_type) |> 
  mutate(mean_mutation_count = mean(as.numeric(mutation_count))) |> 
  ungroup() |> 
  select(cancer_type, cancer_abbrev, mean_mutation_count) |> 
  unique() 
```

```{r bar chart - n mutations for each cancer type, echo = FALSE}
STMN2_cryptic_cBio_mutations |> 
  ggplot(aes(x = fct_reorder(cancer_abbrev, mean_mutation_count, median), y = mean_mutation_count)) +
  geom_bar(stat = 'identity', aes(fill = cancer_abbrev)) +
  labs(
    x = "Cancer Type",
    y = "Mean Mutation Count"
  ) +
  theme(legend.position = "none")
```

## Mutation Data of one patient

```{r data import, include = FALSE}
patient_mutations_orig <- read.csv("TCGA-A5-A0G2_mutations.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE) 

patient_mutations <- patient_mutations_orig |> 
  select(Gene, Protein.Change, Annotation, Functional.Impact, Chromosome, Start.Pos, End.Pos, Ref, Var, HGVSg, Mutation.Type, Variant.Type, Allele.Freq)
```

```{r number of mutations in each gene, echo = FALSE}
patient_mutations |> 
  count(Gene, sort = TRUE) |> 
  filter(n > 17) |> 
  ggplot(aes(x = fct_reorder(Gene, n, mean), y = n)) +
  geom_bar(stat = 'identity', aes(fill = Gene)) +
  labs(
    x = "Gene",
    y = "Mutation Count"
  ) +
  theme(legend.position = "none")
```

```{r types of mutations in TTN, echo = FALSE}
patient_mutations |> 
  filter(Gene == "TTN") |> 
  ggplot(aes(x = fct_rev(fct_infreq(Mutation.Type)))) +
  geom_bar(aes(fill = Mutation.Type)) +
  labs(
    x = "Mutation Type",
    y = "Count in TTN Gene"
  ) +
  theme(legend.position = "none")
```

# TCGA biolinks

```{r include = FALSE}
library(TCGAbiolinks)

    # all cancer abbreviations

cancer_abbrev <- unique(STMN2_cryptic_cBio$cancer_abbrev, na.rm = TRUE)
cancer_abbrev
```
