---
title: "Clinical, gene and mutation data of cancers with cryptic STMN2 events"
output: pdf_document
---

```{r setup, include=FALSE}
options(tinytex.verbose = TRUE)
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(repurrrsive)
library(jsonlite)
library(clipr)
library(naniar)
library(ggpubr)
library(rstatix)
library(ggplot2)
library(tidyr)
library(dplyr)
library(knitr)
library(snapcount)
library(ggsignif)
library(TCGAbiolinks)
library(ggsurvfit)
library(survival)
library(survminer)
```

```{r snapcount_STMN2.R, include = FALSE}
jir_new <- read.csv("jir_new.csv") |> 
  rename("case_submitter_id" = "gdc_cases.submitter_id")
```

```{r GDC case IDs, include = FALSE}
jir_new$case_submitter_id
#write_clip(jir_new$case_submitter_id)
```

```{r Import gene data from TCGA, include = FALSE}
STMN2_events_genes_orig <- read.csv("STMN2_events_genes.tsv", sep = "\t", header = TRUE, na.strings="", fill = TRUE)
head(STMN2_events_genes_orig)

STMN2_events_genes <- STMN2_events_genes_orig

STMN2_events_genes <- STMN2_events_genes |> 
  rename("affected_cases_in_cohort" = "X..SSM.Affected.Cases.in.Cohort") |> 
    # number of cases where gene is mutated / number of cases tested for simple somatic mutations (3,259)
  rename("affected_cases_in_GDC" = "X..SSM.Affected.Cases.Across.the.GDC") |> 
    # number of cases where gene contains simple somatic mutations / number of cases tested for simple somatic mutations portal wide (13,714)
  rename("CNV_gain" = "X..CNV.Gain") |> 
    # number of cases where CNV gain observed in gene / number of cases tested for copy number alteration in gene (3,344)
  rename("CNV_loss" = "X..CNV.Loss") |> 
    # number of cases where CNV loss observed in gene / number of cases tested for copy number alteration in gene (3,344)
  rename("mutations" = "X..Mutations")
    # unique simple somatic mutations in the gene in cohort 
```


```{r Import mutation data from TCGA, include = FALSE}
STMN2_events_mutations_orig <- read.csv("STMN2_events_mutations.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)
head(STMN2_events_mutations_orig)

STMN2_events_mutations <- STMN2_events_mutations_orig
```

```{r Import STMN2 clinical data from TCGA, include = FALSE}
STMN2_clinical_orig <- read.csv("STMN2_clinical.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)
head(STMN2_clinical_orig)

STMN2_clinical <- STMN2_clinical_orig

STMN2_clinical <- STMN2_clinical|> 
  select(case_submitter_id, project_id, age_at_index, ethnicity, gender, race, ajcc_pathologic_stage) |> 
  separate(project_id, into = c("project", "cancer_type")) |> 
  select(-project)

STMN2_clinical_jir <- jir_new |> 
  left_join(STMN2_clinical, by=c("case_submitter_id")) 
STMN2_clinical_jir <- STMN2_clinical_jir |> 
  select(-gdc_cases.diagnoses.tumor_stage, -age_at_index, -gdc_cases.demographic.gender, 
         -ajcc_pathologic_stage, -ethnicity, -race, -gender)
 
STMN2_clinical_jir <- STMN2_clinical_jir |> 
  rename("cancer" = "gdc_cases.project.name") |> 
  rename("gdc_primary_site" = "gdc_cases.project.primary_site") |> 
  rename("sample_type" = "gdc_cases.samples.sample_type") |> 
  rename("cgc_primary_site" = "cgc_case_primary_site") |> 
  rename("cancer_abbrev" = "cancer_type")

# filter for cryptic coverage > 2
STMN2_clinical_jir_cryptic <- STMN2_clinical_jir |> 
  filter(STMN2_cryptic_coverage > 2)
```

# STMN2 junction coverage

### Overall STMN2 junction coverage in different cancer sites

```{r boxplot - overall STMN2 junction coverage in different cancer sites, echo = FALSE, warning = FALSE}
STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  ggplot(aes(x = junction_avg_coverage, y = fct_reorder(gdc_primary_site, junction_avg_coverage, median))) +
  geom_boxplot(aes(fill = gdc_primary_site)) +
  labs(
    x = "Junction Average Coverage",
    y = "Primary Site of Cancer",
  ) +
  theme(legend.position = "none", plot.title = element_text(size=10)) +
  geom_signif(comparisons = list(c("Stomach", "Breast"), c("Stomach", "Brain")),
              map_signif_level = TRUE,
              y_position = c(75, 80))
```
**Figure 1**: Stomach and breast cancers are the most deeply sequenced in general.

### Cryptic STMN2 junction coverage in different cancer sites

```{r cryptic STMN2 junction coverage in different cancer sites, echo = FALSE}
STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  ggplot(aes(x = STMN2_cryptic_coverage, 
             y = fct_reorder(gdc_primary_site, STMN2_cryptic_coverage, median))) +
  geom_boxplot(aes(fill = gdc_primary_site)) +
  labs(
    x = "Cryptic Coverage",
    y = "Primary Site of Cancer"
  ) +
  theme(legend.position = "none", plot.title = element_text(size=10))
```
**Figure 2**: 

```{r adding reads per million (rpm) column for cryptic coverage, echo = FALSE}
STMN2_clinical_jir_cryptic <- STMN2_clinical_jir_cryptic |> 
  mutate(rpm = (STMN2_cryptic_coverage/junction_coverage)*1000000)

STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  ggplot(aes(x = rpm,
             y = fct_reorder(gdc_primary_site, rpm, median))) +
  geom_boxplot(aes(fill = gdc_primary_site)) +
  labs(
    x = "reads per million",
    y = "Primary Site of Cancer"
  ) +
  theme(legend.position = "none")
```
**Figure 3**:

On average, cancers in the stomach and breast have the greatest number of reads supporting cryptic STMN2 events. Cancers of the brain have significantly fewer reads surpporting cryptic events. 

Cancers of the cervix, uterus and liver have the greatest cryptic coverage. However, there is only one data value for cancers of the cervix and liver. 

# Cryptic STMN2 Expression

## Cancers with STMN2 events (in general)

```{r bar plot - cancer types with general STMN2 expression, echo = FALSE}
STMN2_clinical_jir |> 
  distinct() |> 
  drop_na() |> 
  ggplot(aes(x = fct_rev(fct_infreq(cancer_abbrev)))) +
  geom_bar(aes(fill = cancer_abbrev)) +
  coord_flip() +
  labs(
    x = "Cancer Type",
    y = "Number of Cases",
    title = "STMN2 is expressed in mostly breast and brain cancers" 
  ) +
  theme(legend.position = "none", plot.title = element_text(size=10)) 
# BRCA = breast cancer; LGG = low-grade gliomas (brain tumours)
```
**Figure 4**: STMN2 events are found mostly in breast and brain cancer patients. 

```{r bar plot - cancer types with cryptic STMN2 expression, echo = FALSE}
STMN2_clinical_jir_cryptic |> 
  distinct() |> 
  drop_na() |> 
  ggplot(aes(x = fct_rev(fct_infreq(cancer_abbrev)))) +
  geom_bar(aes(fill = cancer_abbrev)) +
  coord_flip() +
  labs(
    x = "Cancer Type",
    y = "Number of Cases"
  ) +
  theme(legend.position = "none", plot.title = element_text(size=10))
```

### Primary sites of cancers 

```{r bar plot - primary sites of cancers with cryptic STMN2 events, echo = FALSE}
STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  ggplot(aes(x = fct_rev(fct_infreq(gdc_primary_site)))) +
  geom_bar(aes(fill = gdc_primary_site)) +
  coord_flip() +
  labs(
    x = "Primary Site of Cancer",
    y = "Number of Cases",
    title = 
  "Cryptic STMN2 events are found mostly in cancers of the adrenal gland and brain" 
  ) +
  theme(legend.position = "none", plot.title = element_text(size=9)) 
```
**Figure 5**: Cryptic STMN2 events are found mostly in the adrenal gland and brain.

Interestingly, Figure 5 shows that cryptic STMN2 expression is low in the breast. Comparing this to Figure 4, this shows that the STMN2 events expressed in breast cancers may be mostly annotated non-cryptic events. 

## Which cancers have the most cryptic STMN2 events?

#### Fraction of each cancer with cryptic STMN2 expression

```{r num / fraction of cases with STMN2 events in cancer types, echo = FALSE}
STMN2_events_different_cancers <- STMN2_clinical_jir_cryptic |> 
  drop_na() |>
  janitor::tabyl(cancer) |> arrange(-percent)

kable(STMN2_events_different_cancers, 
      caption = "Breast and brain cancers have high cryptic STMN2 expression.")

STMN2_events_different_cancers |> 
  mutate(percentage = percent*100) |> 
  ggplot(aes(x = reorder(cancer, percentage), y = percentage)) +
  geom_bar(aes(fill = cancer), stat = "identity") +
  coord_flip() +
  labs(
    x = "Cancer Type",
    y = "Percentage of Cases"
  ) +
  theme(legend.position = "none")
```
**Figure 6**:

## Where are the cancers with cryptic STMN2 events located?

#### Fraction of cases with STMN2 events in cancer sites

```{r number / fraction of cases with STMN2 events in different primary cancer sites, echo = FALSE}
STMN2_events_primary_sites1 <- STMN2_clinical_jir_cryptic |> 
  drop_na() |> 
  janitor::tabyl(gdc_primary_site) |> arrange(-percent)

kable(STMN2_events_primary_sites1, caption = "Cancers with cryptic STMN2 events are found primarily in the adrenal gland and brain")

STMN2_events_primary_sites1 |> 
  mutate(percentage = percent*100) |> 
  ggplot(aes(x = reorder(gdc_primary_site, percentage), y = percentage)) +
  geom_bar(aes(fill = gdc_primary_site), stat = "identity") +
  coord_flip() +
  labs(
    x = "Primary Sites of Cancer",
    y = "Percentage of Cases",
    title = "Cancers with cryptic STMN2 events are primarily in the adrenal gland and brain"
  ) +
  theme(legend.position = "none", plot.title = element_text(size=8))
# 46% of cancers with cryptic STMN2 events are in the adrenal gland
```
**Figure 7**:

# TCGA/cBioPortal Clinical Data

```{r data import - TCGA Clinical Data, include = FALSE}
TCGA_all_clinical_orig <- read.csv("TCGA_all_clinical.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)
head(TCGA_all_clinical_orig)

TCGA_clinical <- TCGA_all_clinical_orig

TCGA_clinical <- TCGA_clinical|> 
  select(case_submitter_id, project_id, age_at_index, ethnicity, gender, race, ajcc_pathologic_stage) |> 
  separate(project_id, into = c("project", "cancer_type")) |> 
  select(-project)
```

```{r data import - cBioPortal Clinical Data, include = FALSE}

cBio_all_clinical_orig <- read.csv("cBio_clinical_data.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE)

cBio_clinical <- cBio_all_clinical_orig

cBio_clinical <- cBio_clinical |> 
  select(Study.ID, Patient.ID, Sample.ID, Aneuploidy.Score, Cancer.Type, TCGA.PanCanAtlas.Cancer.Type.Acronym, Cancer.Type.Detailed,
         Months.of.disease.specific.survival, Mutation.Count, Overall.Survival..Months.,
         Disease.specific.Survival.status)

cBio_clinical <- cBio_clinical |> 
  rename("case_submitter_id" = "Patient.ID") |> 
  rename("cancer" = "Cancer.Type.Detailed") |> 
  rename("cancer_abbrev" = "TCGA.PanCanAtlas.Cancer.Type.Acronym")
```

```{r join cBio clinical data with STMN2 cryptic table, include = FALSE}
STMN2_cryptic_cBio <- STMN2_clinical_jir_cryptic |> 
  left_join(cBio_clinical, by = "case_submitter_id") |> 
  janitor::clean_names()

STMN2_cryptic_cBio <- STMN2_cryptic_cBio |> 
   select(-c(cgc_primary_site, sample_id_2, cancer_abbrev_y, cancer_y, cancer_type)) |> 
  rename("cancer_type" = "cancer_x") |> 
  rename("cancer_abbrev" = "cancer_abbrev_x")
```

## Mutation Counts

```{r calculate mean n mutations for each cancer type, include = FALSE}
STMN2_cryptic_cBio_mutations <- STMN2_cryptic_cBio |> 
  drop_na() |> 
  filter(mutation_count !="NA") |> 
  group_by(cancer_type) |> 
  mutate(mean_mutation_count = mean(as.numeric(mutation_count))) |> 
  ungroup() |> 
  select(cancer_type, cancer_abbrev, mean_mutation_count) |> 
  unique() 
```

```{r bar chart - n mutations for each cancer type, include = FALSE}
STMN2_cryptic_cBio_mutations |> 
  ggplot(aes(x = fct_reorder(cancer_abbrev, mean_mutation_count, median), y = mean_mutation_count)) +
  geom_bar(stat = 'identity', aes(fill = cancer_abbrev)) +
  labs(
    x = "Cancer Type",
    y = "Mean Mutation Count",
    title = "Glioblastoma Multiforme has the greatest number of mutations"
  ) +
  theme(legend.position = "none")

# There is an outlier in GBM for mutation count, which heavily skews the mean for this cancer type. 
```

```{r boxplot - n mutations for each cancer type, echo = FALSE, warning = FALSE}
STMN2_cryptic_cBio <- STMN2_cryptic_cBio |> 
  mutate_at("mutation_count", as.numeric) 

STMN2_cryptic_cBio |> 
  drop_na() |> 
  filter(mutation_count < 2500) |> 
  ggplot(aes(x = fct_reorder(cancer_abbrev, mutation_count, median), y = mutation_count)) +
  geom_boxplot() +
  labs(
    x = "Cancer Type",
    y = "Mutation Count",
    title = "UCEC and LUSC cancers have the greatest number of mutations"
  ) +
  geom_signif(comparisons = list(c("LUSC", "UCEC")),
map_signif_level = TRUE,
y_position = c(1000))
```
**Figure 8**: Among cancer patients with STMN2 cryptic expression, uterine corpus endometrial carcinoma (UCEC) has the greatest number of mutations.

# Mutational burden 

```{r fraction of mutations in each cancer type that is found in cases with cryptic, include = FALSE}
total_mutations_each_cancer <- cBio_clinical |> 
  drop_na(Mutation.Count) |> 
  filter(grepl("^\\d+$", Mutation.Count)) |>
  group_by(cancer_abbrev) |> 
  summarise(total_mutations = sum(as.numeric(Mutation.Count)))

total_mutations_each_cancer_with_cryptic <- STMN2_cryptic_cBio |> 
  drop_na(mutation_count) |> 
  filter(mutation_count < 2500) |> 
  group_by(cancer_abbrev) |> 
  summarise(total_mutations_cryptic = sum(mutation_count))

mutations_each_cancer_general_vs_STMN2cryptic <- total_mutations_each_cancer |> 
  left_join(total_mutations_each_cancer_with_cryptic, by=c("cancer_abbrev")) |> 
  mutate(percent_with_cryptic = (total_mutations_cryptic/total_mutations)*100)
  #16% of all mutations in LGG cancer are in cases with STMN2 cryptic events
```

# Fraction of each cancer that has cryptic STMN2 events 

```{r barplot - fraction of cases of each cancer that have cryptic STMN2, echo = FALSE}
total_each_cancer <- cBio_clinical |> 
  group_by(cancer_abbrev) |> 
  summarise(total_cancer_abbrev = n())

total_each_cancer_with_cryptic <- STMN2_cryptic_cBio |> 
  group_by(cancer_abbrev) |> 
  summarise(total_cancer_abbrev_with_cryptic = n())

total_each_cancer_general_vs_cryptic <- total_each_cancer |> 
  left_join(total_each_cancer_with_cryptic, by=c("cancer_abbrev")) |> 
  mutate(percent_with_cryptic = (total_cancer_abbrev_with_cryptic/total_cancer_abbrev)*100)

total_each_cancer_general_vs_cryptic |> 
  drop_na() |> 
  ggplot(aes(x = reorder(cancer_abbrev, percent_with_cryptic), y = percent_with_cryptic)) +
  geom_bar(aes(fill = cancer_abbrev), stat = "identity") +
  labs(
    x = "Cancer Type",
    y = "Percentage of cases with cryptic STMN2 events"
    ) +
  theme(legend.position = "none")
```
**Figure 9**: 

```{r calculating mutation burden for general vs cryptic, echo = FALSE, warning = FALSE}
cBio_clinical |>
  left_join(STMN2_clinical_jir,by = c("case_submitter_id")) |> 
  janitor::clean_names() |> 
  rename("cancer_abbrev" = "cancer_abbrev_x") |> 
  select(case_submitter_id, study_id, cancer_abbrev, mutation_count, stmn2_cryptic_coverage, stmn2_annotated_coverage) |> 
  separate(study_id,into = ('study_start'),remove = FALSE) |> 
  unique() |> 
  mutate(mutation_count = as.numeric(mutation_count)) |> 
  filter(!is.na(mutation_count) & mutation_count != "NA") |> 
  mutate(stmn2_cryptic_detected = stmn2_cryptic_coverage >= 2) |> 
  group_by(study_start) |> 
  mutate(n_total_samples = n_distinct(case_submitter_id)) |> 
  mutate(n_detected_stmn2 = sum(stmn2_cryptic_detected,na.rm = TRUE)) |> 
  ungroup() |> 
  filter(!is.na(stmn2_cryptic_detected)) |> 
  filter(n_detected_stmn2 > 2) |> 
  ggplot(aes(x = stmn2_cryptic_detected,
             y = mutation_count)) + 
  geom_boxplot(aes(fill = stmn2_cryptic_detected)) + 
  labs(x = "Cancer Type",
       y = "Mutation Count") +
  facet_wrap(~study_start) +
  scale_y_continuous(trans = scales::pseudo_log_trans()) +
  stat_compare_means(comparisons = list(c("TRUE", "FALSE")), label = "p.format")
```
**Figure 10**: 

# Survival Comparisons

```{r, include = FALSE}
survival_STMN2_cryptic <- cBio_clinical |> 
  left_join(STMN2_clinical_jir, by = c("case_submitter_id")) |> 
   janitor::clean_names() |> 
  rename("cancer_abbrev" = "cancer_abbrev_x") |> 
  select(case_submitter_id, study_id, sample_id, cancer_abbrev, mutation_count, stmn2_cryptic_coverage, stmn2_annotated_coverage, months_of_disease_specific_survival, overall_survival_months, disease_specific_survival_status) |> 
  mutate(months_of_disease_specific_survival = as.numeric(months_of_disease_specific_survival)) |> 
  filter(!is.na(months_of_disease_specific_survival) & months_of_disease_specific_survival != "NA") |> 
  mutate(mutation_count = as.numeric(mutation_count)) |> 
  filter(!is.na(mutation_count) & mutation_count != "NA") |> 
  mutate(stmn2_cryptic_detected = stmn2_cryptic_coverage >= 2) |> 
  group_by(cancer_abbrev) |> 
  mutate(n_detected_stmn2 = sum(stmn2_cryptic_detected,na.rm = TRUE)) |>
  ungroup() |> 
  filter(!is.na(stmn2_cryptic_detected)) |> 
  filter(n_detected_stmn2 > 2) |> 
  mutate(stmn2_cryptic_detected = as.logical(stmn2_cryptic_detected)) |> #changes FALSE and TRUE to 0 and 1 respectively
  distinct()
```

```{r survival status column in STMN2 survival df}
survival_STMN2_cryptic <- survival_STMN2_cryptic |> 
  filter(!is.na(disease_specific_survival_status) & disease_specific_survival_status != "NA") |> 
  mutate(disease_specific_survival_status = str_extract(disease_specific_survival_status, "^[^:]+"),
         disease_specific_survival_status = as.numeric(disease_specific_survival_status))
    # 1 = dead with tumor
    # 0 = alive or dead tumor free
```

```{r survival KM curve for cases with STMN2 cryptic, echo = FALSE}
survfit(Surv(months_of_disease_specific_survival, disease_specific_survival_status) ~ 1, 
        data = subset(survival_STMN2_cryptic, stmn2_cryptic_detected==TRUE)) |> 
  ggsurvfit() +
  labs(
    x = "Months",
    y = "Overall Survival Probability",
    title = "Kaplan-Meier Curve: STMN2 Cryptic Events"
  ) +
  add_confidence_interval() 
```
**Figure 11**:

```{r survival KM curve for cases with no STMN2 cryptic, echo = FALSE}
survfit(Surv(months_of_disease_specific_survival, disease_specific_survival_status) ~ 1, 
        data = subset(survival_STMN2_cryptic, stmn2_cryptic_detected==FALSE)) |> 
  ggsurvfit() +
  labs(
    x = "Months",
    y = "Overall Survival Probability",
    title = "Kaplan-Meier Curve: No STMN2 Cryptic Events"
  ) +
  add_confidence_interval() 
```
**Figure 12**:

```{r survival KM curves STMN2 - both cryptic and non, echo = FALSE}
survfit(Surv(months_of_disease_specific_survival, disease_specific_survival_status) ~ stmn2_cryptic_detected, 
        data = survival_STMN2_cryptic) |> 
  ggsurvfit() +
  labs(
    x = "Months with disease",
    y = "Overall Survival Probability",
    title = "Kaplan-Meier Curve: All Cancer Types (STMN2)"
  ) +
  add_confidence_interval() 
```
**Figure 13**: 
  
```{r density plot - months survival in cryptic vs non-cryptic STMN2, echo = FALSE}
survival_STMN2_cryptic |> 
  mutate(stmn2_cryptic_detected = as.logical(stmn2_cryptic_detected)) |> 
  ggplot(aes(x = months_of_disease_specific_survival, colour = stmn2_cryptic_detected)) +
  geom_density() +
  labs(
    x = "Survival months with disease"  
    )
    # same survival - cryptic vs non-cryptic (all cancers combined)
```
**Figure 14**: 

```{r number of STMN2 detected and non in each cancer type, include = FALSE}
STMN2_n_detected_non_cancer_abbrev <- survival_STMN2_cryptic |> 
  group_by(cancer_abbrev, stmn2_cryptic_detected) |> 
  summarise(count = n()) |> 
  ungroup() |> 
  pivot_wider(
    names_from = stmn2_cryptic_detected,
    values_from = count
  ) |> 
  rename("stmn2_cryptic_true" = "TRUE") |> 
  rename("stmn2_cryptic_false" = "FALSE")
      # PCPG has 130 false, 51 true
      # GBM has 154 false, 11 true
```

# Comparing Aneuploidy cancer-by-cancer

```{r boxplot aneuploidy score cancer-by-cancer in cryptic vs none}
aneuploidy_STMN2 <- cBio_clinical |>
  left_join(STMN2_clinical_jir,by = c("case_submitter_id")) |> 
  janitor::clean_names() |> 
  rename("cancer_abbrev" = "cancer_abbrev_x") |> 
  select(case_submitter_id, study_id, cancer_abbrev, aneuploidy_score, stmn2_cryptic_coverage, stmn2_annotated_coverage) |> 
  separate(study_id,into = ('study_start'),remove = FALSE) |> 
  unique() |> 
  mutate(aneuploidy_score = as.numeric(aneuploidy_score)) |> 
  filter(!is.na(aneuploidy_score) & aneuploidy_score != "NA") |> 
  mutate(stmn2_cryptic_detected = stmn2_cryptic_coverage >= 2) |> 
  group_by(study_start) |> 
  mutate(n_total_samples = n_distinct(case_submitter_id)) |> 
  mutate(n_detected_stmn2 = sum(stmn2_cryptic_detected,na.rm = TRUE)) |> 
  ungroup() |> 
  filter(!is.na(stmn2_cryptic_detected)) |> 
  filter(n_detected_stmn2 > 2) |> 
  group_by(study_start) |> 
  mutate(wilcox_result = list(broom::tidy(wilcox.test(aneuploidy_score ~ stmn2_cryptic_detected, exact=FALSE)))) |> 
  add_significance() |> 
  ungroup() |> 
  unnest(wilcox_result)

aneuploidy_STMN2 |> 
  ggplot(aes(x = stmn2_cryptic_detected,
             y = aneuploidy_score)) + 
  geom_boxplot(aes(fill = stmn2_cryptic_detected)) + 
  labs(x = "Cancer Type",
       y = "Aneuploidy Score") +
  facet_wrap(~study_start) +
  stat_compare_means(comparisons = list(c("TRUE", "FALSE")), label = "p.format") +
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.25))) +
  theme(legend.position = "bottom") +
  guides(fill = guide_legend(title = "Cryptic STMN2 detected")) +
  theme(
    axis.title.x = element_text(size = 12, face = "bold"),
    axis.title.y = element_text(size = 12, face = "bold"),
    axis.text.x = element_text(size = 10),
    axis.text.y = element_text(size = 10),
    strip.text = element_text(size = 10, face = "bold"),
   strip.background = element_rect(fill = "lightgray"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12)
  )
```


## Mutation Data of one patient

```{r data import, include = FALSE}
patient_mutations_orig <- read.csv("TCGA-06-5416_mutations.tsv", sep = "\t", header = TRUE, na.strings = "", fill = TRUE) 

patient_mutations <- patient_mutations_orig |> 
  select(Gene, Protein.Change, Annotation, Functional.Impact, Chromosome, Start.Pos, End.Pos, Ref, Var, HGVSg, Mutation.Type, Variant.Type, Allele.Freq)
```

```{r number of mutations in each gene, include = FALSE}
patient_mutations |> 
  count(Gene, sort = TRUE) |> 
  filter(n > 10) |> 
  ggplot(aes(x = fct_reorder(Gene, n, mean), y = n)) +
  geom_bar(stat = "identity", aes(fill = Gene)) +
  labs(
    x = "Gene",
    y = "Mutation Count"
  ) +
  theme(legend.position = "none")
```

```{r types of mutations in TTN, include = FALSE}
patient_mutations |> 
  filter(Gene == "TTN") |> 
  unique() |> 
  ggplot(aes(x = fct_rev(fct_infreq(Mutation.Type)))) +
  geom_bar(aes(fill = Mutation.Type)) +
  labs(
    x = "Mutation Type",
    y = "Count in TTN Gene"
  ) +
  theme(legend.position = "none")
```

## STMN2 cryptic expression 

```{r STMN2 cryptic coverage vs mutation count, echo = FALSE}

STMN2_cryptic_cBio <- STMN2_cryptic_cBio |> 
  mutate_at("mutation_count", as.numeric) 

STMN2_cryptic_cBio |> 
  drop_na() |> 
  ggplot(aes(x = stmn2_cryptic_coverage, y = mutation_count)) +
  labs(
    x = "Number of STMN2 cryptic events",
    y = "Mutation Count"
  ) +
  geom_point()

STMN2_cryptic_cBio |> 
  drop_na() |> 
  filter(mutation_count < 400, stmn2_cryptic_coverage > 10) |> 
  ggplot(aes(x = as.factor(stmn2_cryptic_coverage), y = mutation_count)) +
  labs(
    x = "Number of STMN2 cryptic events",
    y = "Mutation Count"
  ) +
  geom_boxplot()
```

**Figure 17**: There is no correlation between number of STMN2 cryptic events and number of mutations.

# Cancer Driver Genes

```{r COSMIC Cancer Gene Consensus, include = FALSE}
cosmic_cancer_genes <- read.csv("cosmic_cancer_gene_consensus.csv") |> 
  rename("Gene" = "Gene.Symbol")

# filter for cancer driver genes

cosmic_patient_mutations <- filter(patient_mutations, Gene %in% cosmic_cancer_genes$Gene)
```

## Mutations in each cancer gene

```{r number of mutations in each cancer gene, echo = FALSE}
cosmic_patient_mutations |> 
  count(Gene, sort = TRUE) |> 
  filter(n > 11) |> 
  ggplot(aes(x = fct_reorder(Gene, n, mean), y = n)) +
  geom_bar(stat = 'identity', aes(fill = Gene)) +
  labs(
    x = "Cancer Gene",
    y = "Mutation Count"
  ) +
  theme(legend.position = "none")
```

**Figure 18**: Ovarian cancer gene MUC16 has the greatest number of mutations.

```{r types of mutations in MUC16, echo = FALSE}
cosmic_patient_mutations |> 
  filter(Gene == "MUC16") |> 
  ggplot(aes(x = fct_rev(fct_infreq(Mutation.Type)))) +
  geom_bar(aes(fill = Mutation.Type)) +
  labs(
    x = "Mutation Type",
    y = "Count in MUC16 Gene"
  ) +
  theme(legend.position = "none")
```

**Figure 19**: Ovarian cancer gene MUC16 has mostly missense mutations.

## Types of cancer genes

```{r types of cancer genes, echo = FALSE}
cosmic_patient_mutations <- cosmic_patient_mutations |> 
  left_join(cosmic_cancer_genes, by = "Gene") 

cosmic_patient_mutations <- cosmic_patient_mutations |> 
  select(-Entrez.GeneId, -Tier, -Hallmark, -Chr.Band, -Somatic, -Germline, -Tumour.Types.Somatic., 
         -Tumour.Types.Germline., -Cancer.Syndrome, -Molecular.Genetics, -Translocation.Partner, 
         -Other.Germline.Mut, -Other.Syndrome, -Synonyms) 

  # fusion
      # Gene fusions, or translocations, resulting from chromosomal rearrangements are the most common mutation class. 
      #They lead to chimeric transcripts or to deregulation of genes through juxtapositioning of novel promoter or enhancer regions.

  # TSG = tumour suppressor gene

  # oncogene

cosmic_patient_mutations |> 
  drop_na() |> 
  filter(Role.in.Cancer !="") |> 
  ggplot(aes(x = Role.in.Cancer)) +
  geom_bar()
```

```{r types, echo = FALSE}
cosmic_patient_mutations <- cosmic_patient_mutations |> 
  mutate(TSG = ifelse(grepl("TSG", Role.in.Cancer), "yes", "no")) |> 
  mutate(oncogene = ifelse(grepl("oncogene", Role.in.Cancer), "yes", "no")) |> 
  mutate(fusion = ifelse(grepl("fusion", Role.in.Cancer), "yes", "no"))

n_TSG <- cosmic_patient_mutations |> 
  janitor::tabyl(TSG)

kable(n_TSG, caption = "45% of cancer genes with cryptic STMN2 expression are tumour suppressor genes.")

n_oncogene <- cosmic_patient_mutations |> 
  janitor::tabyl(oncogene)

kable(n_oncogene, caption = "39% of cancer genes with cryptic STMN2 expression are oncogenes.")

n_fusion <- cosmic_patient_mutations |> 
  janitor::tabyl(fusion)

kable(n_fusion, caption = "44% of cancer genes with cryptic STMN2 expression are gene fusions.")
```

# TCGA biolinks

```{r patients w cryptic vs patients w/o cryptic - mutations and genes}

```

```{r include = FALSE}
library(TCGAbiolinks)

    # all cancer abbreviations

cancer_abbrev <- unique(STMN2_cryptic_cBio$cancer_abbrev, na.rm = TRUE)
cancer_abbrev
```
